esp/esdllib/README.md

```diff
@@ -0,0 +1,1608 @@
+
+```xml
+<es:set-value target="name" value="'foobar'"/>
+```
+
+In some cases you may see XML or xpath parsing errors when using the single quotes depending on your workflow (_establish precisely when this occurs_). This may be caused by unescaped apostrophes, which you can remedy like this:
+
+```xml
+<es:set-value target="name" value="&apos;foobar&apos;"/>
+```
+
+So far we've concluded that escaping is not required in these attributes:
+
+* `getDataSection()` function inside entry point `@source` attribute
+* `set-value[@value]` attribute
+
+### Implicit Variables
+
+These special-case variables are implicitly defined by the scripting engine. They give scripts access to characteristics of the current transaction and context. For example, say we've defined an ESP service named `WsPersonInfo` with a method named `GetPersonInfo` that has an interface defined as:
+
+```xml
+    <GetPersonInfoRequest>
+        <FirstName/>
+        <LastName/>
+    </GetPersonInfoRequest>
+```
+
+And is bound to a back end with this configuration:
+
+```xml
+    <Method name="GetPersonInfo"  url="https://backend.roxie.com:1234" mode="ESDL" queryname="PersonInfo_Services.GetPersonInfo" status="available"/>
+```
+
+* `$method`
+: The name of the ESP method invoked by the current transaction request. In our example this is the string *GetPersonInfo*.
+
+* `$request`
+: The name of the ESP request structure at the root of the current transaction request. In our example this is the string *GetPersonInfoRequest*.
+
+* `$service`
+: The name of the service containing the current transaction's method. In our example this is the string *WsPersonInfo*.
+
+* `$query`
+: The name of the back end roxie query. In our example this is the string *PersonInfo_Services.GetPersonInfo*.
+
+### System Parameters
+
+Each of these parameters gives the script access to some information about the current execution context. In order to reference a parameter, it must be declared in the script like so:
+
+```xml
+<es:param name="clientversion" select="'unknown'"/>
+```
+
+If the parameter is undefined, then it is assigned the value of the `@select` XPath expression.
+
+* `$clientversion` : The client interface version specified in the incoming request.
+* `$espUserName` : The username of the user authenticated to run the current transaction.
+* `$espUserPeer` : The IP address of host originating the current transaction.
+* `$espUserRealm` : The 'Company' or parent entity to the currently authenticated user.
+* `$espUserStatus` : The status of the currently authenticated user (eg *Active*, *Suspended*, etc., as defined by the security plugin and back end in use)
```

Comment:
I guess you can also add the `espTransactionID` from https://github.com/hpcc-systems/HPCC-Platform/blob/master/esp/esdllib/esdl_script.cpp#L2189

Reply:
Added, and also added some more details about where the Implicit Variables are derived from.